tests/suites/test_suite_pk.function

```diff
@@ -1182,7 +1182,7 @@ exit:
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_USE_PSA_CRYPTO */
+/* BEGIN_CASE depends_on:MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_PK_WRITE_C */
```

Comment:
This is surprising, I thought the point of the PR was removing that dependency for ECDSA. So, I think the dependency on `PK_WRITE` should not be for the entire function, but just for the test cases (in the .data file) that use this function with RSA.

Reply:
I tried to solve this problem in [this commit](https://github.com/Mbed-TLS/mbedtls/pull/6389/commits/e6af138bd5b6cc49ba70c2eb8f3891f0b2f6df14), but now there is a failure in `test_suite_pk` because the function `pk_psa_sign` internally uses `mbedtls_pk_write_pubkey_der` (as wrappers did previously).
However instead of solving this as we did [here](https://github.com/Mbed-TLS/mbedtls/pull/6389/commits/59591db4dc0be0667a123a7b18e0ed8472db2656) I suggest to address your comment in [this issue](https://github.com/Mbed-TLS/mbedtls/issues/6009#issuecomment-1413489731) and fix the problem properly on both sides.
Wdyt?

Reply:
Ah, sorry, I missed that. Clearly if the test function itself calls a function from PK-write then the dependency in this place was justified. It's not that the library functions that we're testing depend on this, it's that the test code depends on it. Ok then. Sorry for missing it in the first place.

Reply:
What if I address this new problem in the same way that was done in `pk_wrap`?
It should go this way:

- in RSA case, we keep the `pk_write` stuff
- in ECDSA one, we go for ECP+BIGNUM

Wdyt?

Reply:
Good idea, let's do this!